src-worker/utils/timeout.ts

```diff
@@ -0,0 +1,31 @@
+export class Timeout {
```

Comment:
What's the purpose of this? I'd try to use utilities built-in nodejs/js  or packages such as lodash instead of making it from scratch. 

See [lodash/throttle](https://lodash.com/docs/4.17.15#throttle)